x/committee/types/codec.go

```diff
@@ -42,26 +25,64 @@ func RegisterCodec(cdc *codec.Codec) {
 
 	// Permissions
 	cdc.RegisterInterface((*Permission)(nil), nil)
-	cdc.RegisterConcrete(GodPermission{}, "kava/GodPermission", nil)
-	cdc.RegisterConcrete(SimpleParamChangePermission{}, "kava/SimpleParamChangePermission", nil)
-	cdc.RegisterConcrete(TextPermission{}, "kava/TextPermission", nil)
-	cdc.RegisterConcrete(SoftwareUpgradePermission{}, "kava/SoftwareUpgradePermission", nil)
-	cdc.RegisterConcrete(SubParamChangePermission{}, "kava/SubParamChangePermission", nil)
+	// cdc.RegisterConcrete(GodPermission{}, "kava/GodPermission", nil)
+	// cdc.RegisterConcrete(SimpleParamChangePermission{}, "kava/SimpleParamChangePermission", nil)
+	// cdc.RegisterConcrete(TextPermission{}, "kava/TextPermission", nil)
+	// cdc.RegisterConcrete(SoftwareUpgradePermission{}, "kava/SoftwareUpgradePermission", nil)
+	// cdc.RegisterConcrete(SubParamChangePermission{}, "kava/SubParamChangePermission", nil)
 
 	// Msgs
-	cdc.RegisterConcrete(MsgSubmitProposal{}, "kava/MsgSubmitProposal", nil)
-	cdc.RegisterConcrete(MsgVote{}, "kava/MsgVote", nil)
+	// cdc.RegisterConcrete(MsgSubmitProposal{}, "kava/MsgSubmitProposal", nil)
+	// cdc.RegisterConcrete(MsgVote{}, "kava/MsgVote", nil)
 }
 
-// RegisterPermissionTypeCodec allows external modules to register their own permission types on
-// the internal ModuleCdc. This allows the MsgSubmitProposal to be correctly Amino encoded and
-// decoded (when the msg contains a CommitteeChangeProposal).
-func RegisterPermissionTypeCodec(o interface{}, name string) {
-	ModuleCdc.RegisterConcrete(o, name, nil)
+func RegisterInterfaces(registry types.InterfaceRegistry) {
+	registry.RegisterInterface(
+		"kava.committee.v1beta1.Committee",
+		(*Committee)(nil),
+		// TODO: Question: Might not need the base one since we just have token and member?
+		&BaseCommittee{},
+		&TokenCommittee{},
+		&MemberCommittee{},
+	)
+
+	// TODO: QUESTION: Is this needed?
+	registry.RegisterImplementations(
```

Comment:
any ideas?

Reply:
Looks like it's not needed.
The `registry` is created in https://github.com/Kava-Labs/kava/blob/ccd7807b7ce759d69b8005d325fb25e00c0a315b/app/params/encoding.go#L22 and passed down to the modules.
Looking at the cosmos code for the registry `RegisterInterface` calls `RegisterImplementations` underneath. https://github.com/cosmos/cosmos-sdk/blob/b75c29fc15d3320ec0c7596dbd7c787c48dccad8/codec/types/interface_registry.go#L98-L117 so I think we don't need it.